x-pack/test/apm_api_integration/common/utils/create_and_run_apm_ml_jobs.ts

```diff
@@ -5,11 +5,35 @@
  * 2.0.
  */
 
+import { Client } from '@elastic/elasticsearch';
 import job from '@kbn/ml-plugin/server/models/data_recognizer/modules/apm_transaction/ml/apm_tx_metrics.json';
 import datafeed from '@kbn/ml-plugin/server/models/data_recognizer/modules/apm_transaction/ml/datafeed_apm_tx_metrics.json';
 import { MlApi } from '../../../functional/services/ml/api';
 
-export function createAndRunApmMlJob({ ml, environment }: { ml: MlApi; environment: string }) {
+export async function createAndRunApmMlJobs({
+  es,
+  ml,
+  environments,
+}: {
+  es: Client;
+  ml: MlApi;
+  environments: string[];
+}) {
+  if (environments.length === 0) {
+    return;
+  }
+
+  // Creating multiple ml jobs in parallel is causing this tests to be flaky
+  // https://github.com/elastic/elasticsearch/issues/36271
+  await createAndRunApmMlJob({ environment: environments.shift() as string, ml });
+  await es.cluster.health({ index: '.ml-state-*', wait_for_status: 'yellow' });
+
+  return await Promise.all(
+    environments.map((environment) => createAndRunApmMlJob({ environment, ml }))
+  );
+}
```

Comment:
I don't think you need this method `createAndRunApmMlJobs`. Afaict it's sufficient to add the cluster health check in `createAndRunApmMlJob`.

From the ES documentation:

> The cluster health status is: green, yellow or red. On the shard level, a red status indicates that the specific shard is not allocated in the cluster, yellow means that the primary shard is allocated but replicas are not, and green means that all shards are allocated. The index level status is controlled by the worst shard status. The cluster status is controlled by the worst index status.

> One of the main benefits of the API is the ability to wait until the cluster reaches a certain high water-mark health level. For example, the following will wait for 50 seconds for the cluster to reach the yellow level (if it reaches the green or yellow status before 50 seconds elapse, it will return at that point).

This means that waiting for "yellow" will also return if the cluster state is green.

Reply:
As discussed, I prefer to have the method so every time we need to create multiple mlJobs we don't forget about not doing it in parallel